frontend/app/services/subscription-service.server.ts

```diff
@@ -75,6 +81,55 @@ export const getSubscriptionService = moize(createSubscriptionService, { onCache
 function createSubscriptionService() {
   const { CDCP_API_BASE_URI } = getEnv();
 
+  /**
+   *
+   * @param userId
+   * @returns the subscription details for the user or null if no user is found or the user has no CDCP subscriptions.
+   */
+  async function updateUser(userId: string, userInfo: UserInfo) {
```

Comment:
I don't think we should be fetching the user data every time we need to update it.

The user data should be fetched at login and stored in session.